src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -673,10 +673,7 @@ bool emberAfOperationalCredentialsClusterAttestationRequestCallback(app::Command
         // TODO: retrieve vendor information to populate the fields below.
         uint32_t timestamp = 0;
         ByteSpan firmwareInfo;
-        ByteSpan * vendorReservedArray = nullptr;
-        size_t vendorReservedArraySize = 0;
-        uint16_t vendorId              = 0;
-        uint16_t profileNum            = 0;
+        Credentials::DeviceAttestationVendorReservedConstructor vendorReserved(nullptr, 0);
```

Comment:
Suggest making it clear that this is for emptiness purposes (if suggestion accepted, will need the renames elsewhere)

```suggestion
        const Credentials::DeviceAttestationVendorReservedConstructor kEmptyVendorReserved(nullptr, 0);
```